commit 122181d964cffacbf7bf639275f77781a85bd8fc
Author: Liu Xiang <liuxiang@ktjr.com>
Date:   Wed Apr 6 14:35:50 2022 +0800

    Make keybindings be more reasonable

diff --git a/Action.c b/Action.c
index 07e21dc..cd081af 100644
--- a/Action.c
+++ b/Action.c
@@ -5,6 +5,7 @@ Released under the GNU GPLv2+, see the COPYING file
 in the source distribution for its full text.
 */
 
+#include "Panel.h"
 #include "config.h" // IWYU pragma: keep
 
 #include "Action.h"
@@ -689,7 +690,7 @@ void Action_setBindings(Htop_Action* keys) {
    keys['c'] = actionTagAllChildren;
    keys['e'] = actionShowEnvScreen;
    keys['h'] = actionHelp;
-   keys['k'] = actionKill;
+   keys[KEY_CTRL('K')] = actionKill;
    keys['l'] = actionLsof;
    keys['m'] = actionToggleMergedCommand;
    keys['p'] = actionToggleProgramPath;
diff --git a/CategoriesPanel.c b/CategoriesPanel.c
index 30867ee..9c05a7f 100644
--- a/CategoriesPanel.c
+++ b/CategoriesPanel.c
@@ -109,6 +109,8 @@ static HandlerResult CategoriesPanel_eventHandler(Panel* super, int ch) {
       case KEY_CTRL('P'):
       case KEY_DOWN:
       case KEY_CTRL('N'):
+      case 'j':
+      case 'k':
       case KEY_NPAGE:
       case KEY_PPAGE:
       case KEY_HOME:
diff --git a/ColumnsPanel.c b/ColumnsPanel.c
index a1450bb..751d602 100644
--- a/ColumnsPanel.c
+++ b/ColumnsPanel.c
@@ -55,13 +55,13 @@ static HandlerResult ColumnsPanel_eventHandler(Panel* super, int ch) {
          }
          break;
       }
-      case KEY_UP:
-      {
-         if (!this->moving) {
-            break;
-         }
+      case 'k':
+      case KEY_UP: {
+        if (!this->moving) {
+          break;
+        }
       }
-         /* else fallthrough */
+        /* else fallthrough */
       case KEY_F(7):
       case '[':
       case '-':
@@ -71,13 +71,13 @@ static HandlerResult ColumnsPanel_eventHandler(Panel* super, int ch) {
          result = HANDLED;
          break;
       }
-      case KEY_DOWN:
-      {
-         if (!this->moving) {
-            break;
-         }
+      case 'j':
+      case KEY_DOWN: {
+        if (!this->moving) {
+          break;
+        }
       }
-         /* else fallthrough */
+        /* else fallthrough */
       case KEY_F(8):
       case ']':
       case '+':
diff --git a/OpenFilesScreen.c b/OpenFilesScreen.c
index 34367eb..416f79d 100644
--- a/OpenFilesScreen.c
+++ b/OpenFilesScreen.c
@@ -119,7 +119,7 @@ static OpenFiles_ProcessData* OpenFilesScreen_getProcessData(pid_t pid) {
       close(fdnull);
       char buffer[32] = {0};
       xSnprintf(buffer, sizeof(buffer), "%d", pid);
-      execlp("lsof", "lsof", "-P", "-o", "-p", buffer, "-F", NULL);
+      execlp("lsof", "lsof", "-P", "-n", "-o", "-p", buffer, "-F", NULL);
       exit(127);
    }
    close(fdpair[1]);
diff --git a/Panel.c b/Panel.c
index a5773d5..02aae20 100644
--- a/Panel.c
+++ b/Panel.c
@@ -350,6 +350,7 @@ bool Panel_onKey(Panel* this, int key) {
    } while (0)
 
    switch (key) {
+   case 'j':
    case KEY_DOWN:
    case KEY_CTRL('N'):
    #ifdef KEY_C_DOWN
@@ -358,6 +359,7 @@ bool Panel_onKey(Panel* this, int key) {
       this->selected++;
       break;
 
+   case 'k':
    case KEY_UP:
    case KEY_CTRL('P'):
    #ifdef KEY_C_UP
@@ -367,7 +369,6 @@ bool Panel_onKey(Panel* this, int key) {
       break;
 
    case KEY_LEFT:
-   case KEY_CTRL('B'):
       if (this->scrollH > 0) {
          this->scrollH -= MAXIMUM(CRT_scrollHAmount, 0);
          this->needsRedraw = true;
@@ -375,15 +376,16 @@ bool Panel_onKey(Panel* this, int key) {
       break;
 
    case KEY_RIGHT:
-   case KEY_CTRL('F'):
       this->scrollH += CRT_scrollHAmount;
       this->needsRedraw = true;
       break;
 
+   case KEY_CTRL('B'):
    case KEY_PPAGE:
       PANEL_SCROLL(-(this->h - Panel_headerHeight(this)));
       break;
 
+   case KEY_CTRL('F'):
    case KEY_NPAGE:
       PANEL_SCROLL(+(this->h - Panel_headerHeight(this)));
       break;
